### PR TITLE
feature/CP-4039-android-geo-fences-with-time-limits_v1

### PR DESCRIPTION
**Pull request by cpKevinrodriguesnobre**
CP-4039 - not getting state trantition state

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:1023]: L1008 where you call addGeofences
maybe do it like here and see if it helps
so add the pendingIntent to the parameters of addGeofences and add the success and failurelistener and ofc some logs inside those
@ozamanav  @jprajapati1996 

image

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1023]: https://developer.android.com/training/location/geofencing#define-a-broadcast-receiver-for-geofence-transitions

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:965]: okay i will comment this line

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java
Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1074]: and does it work now? or what is the situation now?

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1074]: also you changed way more than i was asking for and the pendingintent already existed here https://github.com/cleverpush/cleverpush-android-sdk/pull/85/commits/d53e6baa6e9f3fa8da9bba5af0928f9d15cb9168#diff-dadb76999f5ed7d5f6c363d011ef4b354183a8b5cee4ecd118f2152c882891f1R1068
please check with manav if you didnt yet

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:49]: Remove comments

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:55]: Remove comments

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:52]: Remove comments

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:67]: Remove comments + new line

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:29]: Remove comments + new line

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:34]: Remove comments + new line

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:115]: add new line at EOF

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:102]: Use `Logger` class

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:38]: Use `Logger` class and no "CPErrorMessage needed

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:32]: Rename to countDownTimer

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:71]: magic number

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:73]: onTick not used - needs to be overridden?

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:88]: Remove string CPJSONException

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:97]: Add Logger.e log here for failure

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:93]: Add Logger.d debug log here

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:45]: Magic strings. Constants or ENUM

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:115]: Added

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:34]: okay removed comment and added new line

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:49]: removed

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:102]: okay used Logger class

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:38]: okay removed Log

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:32]: okay renamed

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:88]: okay removed only string

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:97]: `Logger.e("", response);`
Added

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:93]: ` Logger.d("onSuccess", response);`
Added

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:62]: Okay changed

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:45]: Added 
` String ENTER = "enter";
    String EXIT = "exit";`

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:31]: `geofenceIndex`

File: cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/GeofenceBroadcastReceiver.java:30]: `geoFenceTimeoutCompleted`

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:1031]: not needed so i removed

**Review by gbhrdt — APPROVED** [review]

**Comment by gbhrdt** [comment]
Waiting for review from @cpKevinrodriguesnobre

File: cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:54]: copy paste mistake?

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:54]: + resolve conflicts

Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/service/CleverPushGeofenceTransitionsIntentService.java:54]: Yes sorry it's copy-paste mistake 
and now all the conflicts are resolved

**Review by cpKevinrodriguesnobre — COMMENTED** [review]
Approved

**Pull request merged**
